src/components/common/Directors.vue

```diff
@@ -1341,70 +1288,32 @@ export default class Directors extends Mixins(CommonMixin, DateMixin, DirectorMi
     return (director.isDirectorActionable !== undefined) ? director.isDirectorActionable : true
   }
 
-  /** Called when there is a change (ie, some <-> none) in directors with a paid change. */
-  @Watch('directorsPaidChange')
-  private onDirectorsChange (val: boolean): void {
-    // emit event back up to parent
-    this.emitDirectorsChange(val)
-  }
-
-  /** Called when there is a change (ie, some <-> none) in directors with a free change. */
-  @Watch('directorsFreeChange')
-  private onDirectorsFreeChange (val: boolean): void {
-    // emit event back up to parent
-    this.emitDirectorsFreeChange(val)
-  }
-
   /** Called when a director form's validity changes. */
   @Watch('directorFormValid')
-  private onDirectorFormValid (val: boolean): void {
+  private onDirectorFormValidChanged (): void {
     // emit event back up to parent
-    this.emitDirectorFormValid(val)
+    this.emitDirectorFormValid()
   }
 
-  /** Called when As Of Date is set or changes (from parent component). */
-  @Watch('asOfDate')
```

Comment:
Again, architecture.

It should not be THIS component that monitors whether to re-fetch the original directors and whether to update the working copy; it should be the parent.

Reply:
To be clear, THIS component should concern itself with displaying/editing its specific data, not high-level logic. (I think even the fetch should not be in this component but in the parent or some common mixin/service.)

Reply:
PS I refactored AR/COA/COA (parents) and Directors/OfficeAddresses (children) to implement the correct architecture.